### PR TITLE
topo: add concrete sized, aligned, topology objs

### PR DESCRIPTION
**Pull request by lheeger-jump**
Adds an explicit function for adding sized and aligned object to topology. This allows us to add objects like blockstore and funk to a topology without special case logic for every single object in a topology.

### CODE REVIEW

File: src/app/fdctl/configure/workspace.c [hunk 1]
```diff
@@ -48,7 +48,14 @@ fdctl_obj_new( fd_topo_t const *     topo,
   } else if( FD_UNLIKELY( !strcmp( obj->name, "metrics" ) ) ) {
     fd_metrics_new( laddr, VAL("in_cnt"), VAL("out_cnt") );
   } else {
-    FD_LOG_ERR(( "unknown object `%s`", obj->name ));
+    /* if the object specified the properities necessary to be concrete */
+    ulong align = fd_pod_queryf_ulong( topo->props, ULONG_MAX, "obj.%lu.align", obj->id );
```
Comment (mmcgee-jump) [inline src/app/fdctl/configure/workspace.c:52]: ??? This does not new the object. Who is going to call new?

File: src/app/fdctl/config.c [hunk 2]
```diff
@@ -559,9 +564,10 @@ fdctl_obj_align( fd_topo_t const *     topo,
   } else if( FD_UNLIKELY( !strcmp( obj->name, "metrics" ) ) ) {
     return FD_METRICS_ALIGN;
   } else {
-    FD_LOG_ERR(( "unknown object `%s`", obj->name ));
-    return 0UL;
+    ulong align = VAL("align");
```
Comment (mmcgee-jump) [inline src/app/fdctl/config.c:567]: This should happen before the other checks... a/k/a at the top of this function.  If the object already specifies a concrete align, then we don't need to calculate it.

File: src/app/fdctl/config.c [hunk 3]
```diff
@@ -596,8 +602,8 @@ fdctl_obj_footprint( fd_topo_t const *     topo,
   } else if( FD_UNLIKELY( !strcmp( obj->name, "metrics" ) ) ) {
     return FD_METRICS_FOOTPRINT( VAL("in_cnt"), VAL("out_cnt") );
   } else {
-    FD_LOG_ERR(( "unknown object `%s`", obj->name ));
-    return 0UL;
+    ulong sz = VAL("sz");
```
Comment (mmcgee-jump) [inline src/app/fdctl/config.c:605]: Same as above

File: src/app/fdctl/config.c [hunk 4]
```diff
@@ -615,6 +621,11 @@ fdctl_obj_loose( fd_topo_t const *     topo,
     }
     fd_topo_run_tile_t * config = fd_topo_tile_to_config( tile );
     if( FD_LIKELY( config->loose_footprint ) ) return config->loose_footprint( tile );
+  } else {
+    ulong loose_sz = fd_pod_queryf_ulong( topo->props, ULONG_MAX, "obj.%lu.%s", obj->id, "loose" );
```
Comment (mmcgee-jump) [inline src/app/fdctl/config.c:625]: As aobve

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
.

File: src/app/fdctl/configure/workspace.c (hunk 1 above)
Comment (lheeger-jump) [reply src/app/fdctl/configure/workspace.c:52]: The user.

Comment (lheeger-jump) [reply src/app/fdctl/configure/workspace.c:52]: it explicitly doesnt happen here. It is your responsibility to set the object up where you actually intend to use it.

File: src/app/fdctl/config.c (hunk 2 above)
Comment (lheeger-jump) [reply src/app/fdctl/config.c]: done

File: src/app/fdctl/config.c (hunk 3 above)
Comment (lheeger-jump) [reply src/app/fdctl/config.c]: done

File: src/app/fdctl/config.c (hunk 4 above)
Comment (lheeger-jump) [reply src/app/fdctl/config.c]: done

File: src/app/fdctl/configure/workspace.c (hunk 1 above)
Comment (lheeger-jump) [reply src/app/fdctl/configure/workspace.c]: Added a better comment there.

**Review by mmcgee-jump — APPROVED** [review]

**Pull request merged**